sycl/source/esimd_emulator_device_interface.cpp

```diff
@@ -28,8 +28,16 @@ __SYCL_EXPORT ESIMDDeviceInterface *getESIMDDeviceInterface() {
   // tight loop)
   void *PIOpaqueData = nullptr;
 
-  PIOpaqueData =
-      getPluginOpaqueData<cl::sycl::backend::ext_intel_esimd_emulator>(nullptr);
+  try {
+    PIOpaqueData =
+        getPluginOpaqueData<cl::sycl::backend::ext_intel_esimd_emulator>(
+            nullptr);
+  } catch (...) {
+    std::cerr << "ESIMD EMU plugin error or not loaded - try setting "
+                 "SYCL_DEVICE_FILTER=esimd_emulator:gpu environment variable"
+              << std::endl;
```

Comment:
Please, do not print messages unconditionally. This message can be in the exception which is thrown below, so a user can read it by calling .what() . Or it can be done under pi trace check, see https://github.com/intel/llvm/blob/sycl/sycl/source/device_selector.cpp#L48

Reply:
Will do.